docs/csharp/quick-starts/branches-and-loops-local.md

```diff
@@ -0,0 +1,305 @@
+executing those statements until the condition is false.
+
+There's one other new operator in this example. The `++` after
+the `counter` variable is the **increment** operator. It adds 1
+to the value of counter, and stores that value in the counter variable.
+
+> [!IMPORTANT]
+> Make sure that the `while` loop condition does switch to
+> false as you execute the code. Otherwise, you create an
+> **infinite loop** where your program never ends. Let's
+> not demonstrate that, because you have to force your program to
+> quit using **CTRL-C** or other means.
+
+The `while` loop tests the condition before executing the code
+following the `while`. The `do` ... `while` loop executes the 
+code first, and then checks the condition. It looks like this:
+
+```csharp
+counter = 0;
+do
+{
+  Console.WriteLine($"Hello World! The counter is {counter}");
+  counter++;
+} while (counter < 10);
+```
+
+This `do` loop and the earlier `while` loop work the same. 
+
+Let's move on to one last loop statement.
+
+## Work with the for loop
+
+Another common loop statement that you'll see in C# code is the
+`for` loop. Try this code in the interactive window:
+
+```csharp
+for(int index = 0; index < 10; index++)
+{
+  Console.WriteLine($"Hello World! The index is {index}");
+} 
+```
+
+This does the same work as the `while` loop and the `do` loop you've
+already used. The `for` statement has three parts that control
+how it works. 
+
+The first part is the **for initializer**: `for index = 0;` declares
+that `index` is the loop variable, and sets its initial value to `0`.
+
+The middle part is the **for condition**: `index < 10` declares that this
+`for` loop continues to execute as long as the value of counter is less than 10.
+
+The final part is the **for iterator**: `index++` specifies how to modify the loop
+variable after executing the block following the `for` statement. Here, it specifies
+that `index` should be incremented by 1 each time the block executes.
+
+Experiment with these yourself. Try each of the following:
+
+- Change the initializer to start at a different value.
+- Change the condition to stop at a different value.
+
+When you're done, let's move on to write some code yourself to
+use what you've learned.
+
+## Combine branches and loops
+
+Now that you've seen the `if` statement and the looping
+constructs in the C# language, see if you can write C# code to
+find the sum of all integers 1 through 20 that are divisible
+by 3.  Here are a few hints:
+
+- The `%` operator gives you the remainder of a division operation.
+- The `if` statement givesx you the condition to see if a number should be part of the sum.
+- The `for` loop can help you repeat a series of steps for all the numbers 1 through 20.
+
+Try it yourself. Then check how you did. You can see one possible answer by
+[looking at the finished sample code on GitHub](https://github.com/dotnet/docs/tree/master/samples/csharp/branches-quickstart/Program.cs#L46-L54).
```

Comment:
`[looking at the finished sample code on GitHub]`
`[viewing the completed code on GitHub]`
or
`[viewing the completed code]`